evalml/tests/component_tests/test_components.py

```diff
@@ -825,6 +825,33 @@ def test_all_estimators_check_fit(X_y_binary, test_estimator_needs_fitting_false
         component.feature_importance
 
 
+@pytest.mark.parametrize("data_type", ['li', 'np', 'pd', 'ww'])
+def test_all_transformers_check_fit_input_type(data_type, X_y_binary, make_data_type):
```

Comment:
@angela97lin FYI for when you start #1406 , we'll already have this coverage in place if this is merged first 🥳